src/models/hit-types/text-hit.ts

```diff
@@ -0,0 +1,198 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Memoize } from 'typescript-memoize';
+import { BooleanField } from '../metadata-fields/field-types/boolean';
+import { DateField } from '../metadata-fields/field-types/date';
+import { MediaTypeField } from '../metadata-fields/field-types/mediatype';
+import { NumberField } from '../metadata-fields/field-types/number';
+import { StringField } from '../metadata-fields/field-types/string';
+
+/**
+ * A model that describes a textual hit from the FTS API.
+ *
+ * The fields in here get casted to their respective field types. See `metadata-fields/field-type`.
+ *
+ * Add additional fields as needed.
+ *
+ * @export
+ * @class TextHit
+ */
+export class TextHit {
```

Comment:
feels like we can make a generic interface/class to show what default values a hit always/usually has...

Reply:
Fulfilled in part by the Result (formerly Hit) type, which includes the full gamut of fields possible across any type of hit (at present, only ItemHit or TextHit). Depending on the type of hit, many of these will actually be undefined -- but that's fine in this case because almost every field on these hits is potentially undefined anyway (since so many of them are optional). The caller is expected to check for the presence of fields before using them, and can always check the hit schema if they truly need to know what type of hit they have.

I guess that's not quite the "always/usual default values" interface you're getting at though. Maybe other solutions here?